### ISSUE TITLE
something wrong in sm.iou / sm.fscore

### ISSUE THREAD

**Issue by nightandweather** [qubvel/segmentation_models#377]
`model = sm.Unet(BACKBONE, encoder_weights='imagenet')
model.compile(
        'Adam',
        loss=sm.losses.Diceloss(),
        metrics=[sm.metrics.IOUScore(threshold = 0.5),sm.metrics.FScore(threshold = 0.5)]
    )
model.load_weights(r"C:\Users\sniperlkh\Desktop\logs\kanghoun_unet_efficientnetb0_6fold_1_20200724-101611\unet_kanghoun_epoch_062.h5")
result_dict = {}
for patient_folder in os.listdir(BASE_PATIENT_LIB):
    IMAGE_TEST_LIB = BASE_PATIENT_LIB +patient_folder + r'/2dimage/'
    MASK_TEST_LIB = BASE_PATIENT_LIB+patient_folder+ r'/2dmask/'
    LungMASK_TEST_LIB = BASE_PATIENT_LIB+patient_folder+r'/2dlungmask/'
    
    import natsort
    all_images = [x for x in sorted(natsort.natsorted(os.listdir(IMAGE_TEST_LIB))) if x[-4:] == '.png']
    all_images = natsort.natsorted(all_images)
    x_data_list = []
    x_data_test = np.empty((len(all_images), IMG_HEIGHT, IMG_WIDTH,3), dtype='float32')
    for i, name in tqdm(enumerate(all_images)):
#         print(name)
        im = cv2.imread(IMAGE_TEST_LIB + name, cv2.IMREAD_UNCHANGED).astype("int16").astype('float32')
        im = cv2.resize(im, dsize=(IMG_WIDTH, IMG_HEIGHT), interpolation=cv2.INTER_LANCZOS4)
        im = (im - np.min(im)) / (np.max(im) - np.min(im))
        x_data_test[i] = im[:,:,0:3]
        x_data_list.append(name)
    all_images = [x for x in sorted(natsort.natsorted(os.listdir(MASK_TEST_LIB))) if x[-4:] == '.png']
    all_images = natsort.natsorted(all_images)
    y_data_test = np.empty((len(all_images), IMG_HEIGHT, IMG_WIDTH,3), dtype='float32')
    y_data_list = []
    for i, name in tqdm(enumerate(all_images)):

        im = cv2.imread(MASK_TEST_LIB + name, cv2.IMREAD_UNCHANGED).astype('float32')/255.
        im = cv2.resize(im, dsize=(IMG_WIDTH, IMG_HEIGHT), interpolation=cv2.INTER_NEAREST)
        y_data_test[i] = im[:,:,0:3]
        y_data_list.append(name)
        
    all_images = [x for x in sorted(natsort.natsorted(os.listdir(MASK_TEST_LIB))) if x[-4:] == '.png']
    all_images = natsort.natsorted(all_images)
    y_lung_data_test = np.empty((len(all_images), IMG_HEIGHT, IMG_WIDTH,3), dtype='float32')
    y_lung_data_list = []
    for i, name in tqdm(enumerate(all_images)):

        im = cv2.imread(MASK_TEST_LIB + name, cv2.IMREAD_UNCHANGED).astype('float32')/255.
        im = cv2.resize(im, dsize=(IMG_WIDTH, IMG_HEIGHT), interpolation=cv2.INTER_NEAREST)
        y_data_test[i] = im[:,:,0:3]
        y_data_list.append(name)
        
    y_hat = model.predict(x_data_test)
    loss,iou_score,f1_score = model.evaluate(
       x_data_test,
       y_data_test)
    result_dict[patient_folder]=loss,iou_score,f1_score
    print(loss,iou_score,f1_score)`

hello, thank you for sharing the results of your efforts qubvel,
I am doing my task easily with your github.

but I have a problem.
my task is find lung nodule in patient ct scan, so I train the model and use keras, model.eval / model.predict to predict in each patient ct scan

I use sm.losses.bce_diceloss , sm.IOU , sm.FScore, however when i use model.eval, it has 0.5 dice score but model.predict, there is no true positive in output..

am i doing right?
